risc0/zkvm/platform/src/memory.rs

```diff
@@ -54,17 +54,14 @@ impl Region {
 }
 
 // These should match the linker script in `risc0/build/risc0.ld`.
-pub const STACK: Region = Region::new(0x0000_0000, mb(16));
-pub const DATA: Region = Region::new(0x0100_0000, mb(16));
-pub const HEAP: Region = Region::new(0x0200_0000, mb(32));
-pub const INPUT: Region = Region::new(0x0400_0000, mb(32));
-pub const PROG: Region = Region::new(0x0600_0000, mb(32));
+pub const INPUT: Region = Region::new(0x0000_0400, mb(32) - kb(1));
+pub const STACK: Region = Region::new(0x0200_0000, mb(16));
+pub const DATA: Region = Region::new(0x0300_0000, mb(16));
+pub const HEAP: Region = Region::new(0x0400_0000, mb(32));
+pub const TEXT: Region = Region::new(0x0600_0000, mb(32));
 pub const OUTPUT: Region = Region::new(0x0800_0000, mb(32));
```

Comment:
Is there any reason to have a specific output + commit region anymore?  As far as I can tell there is no reason output can't just send heap pointers to the host.  Also for the commits, cant we just have a single SHA256 state and a single 16 word staging buffer (both in the heap), and just write words to the staging buffer and kick off a compress whenever it's full?

Reply:
Ah yes, that's a good point on both counts. Now that we have `sys_sha_buffer`, we can probably drop the COMMIT region. I also don't think we need an OUTPUT region anymore!